### PR TITLE
[GEOS-9205] Failure to encoding GeoJSON when returning a complex feature out of the single feature WFS3 endpoint

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: src/community/wfs3/src/test/java/org/geoserver/wfs3/ComplexFeatureTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/*
```
Comment (nmco) [inline src/community/wfs3/src/test/java/org/geoserver/wfs3/ComplexFeatureTest.java]: Wrong project :yum:

File: src/community/wfs3/pom.xml [hunk 2]
```diff
@@ -145,6 +145,43 @@
       <artifactId>jsoup</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.geotools</groupId>
+      <artifactId>gt-app-schema</artifactId>
+      <version>${gt.version}</version>
+    </dependency>
```
Comment (nmco) [inline src/community/wfs3/pom.xml]: So WFS 3 depends on App-Schema now? Just making sure this is not a forgotten dependency.

**Comment by nmco** [comment]
Just some small comments, otherwise looks good.

File: src/community/wfs3/pom.xml (hunk 2 above)
Comment (aaime) [reply src/community/wfs3/pom.xml]: Needed to run a request based on app-schema... ah, but you're right, could have been a test dependency. Fixing.

File: src/community/wfs3/src/test/java/org/geoserver/wfs3/ComplexFeatureTest.java (hunk 1 above)
Comment (aaime) [reply src/community/wfs3/src/test/java/org/geoserver/wfs3/ComplexFeatureTest.java]: Right, intellij automatic headers hit again.

**Comment by aaime** [comment]
Updated

**Pull request merged**